### ISSUE TITLE
Heading Block: Allow quickly adding specific heading levels

### ISSUE THREAD

**Issue by shaunandrews** [WordPress/gutenberg#25318]
Right now, to add an h4 to my document I need to add a Heading block and then use the block's toolbar to select the h4 level from the menu.

<img width="450" alt="image" src="https://user-images.githubusercontent.com/191598/93139589-09106300-f6af-11ea-9c41-415cc59123fe.png">

I often add the Heading block from the `/` inserter:

<img width="356" alt="image" src="https://user-images.githubusercontent.com/191598/93139665-30673000-f6af-11ea-8ebc-67a41cc01c9f.png">

What I'd _love_ to do, is to type `/h4` and have the Heading block added with its level set to h4. I _think_ we could do this by creating hidden variations of the Heading block that would only appear when searching.

**Comment by diegohaz** [comment]
That would be really good!

I used to type `/head` + <kbd>Enter</kbd> as well. But since I always had to change the heading level manually later I trained myself to use `####` + <kbd>Enter</kbd>  (which would create a heading level 4).

I would like to keep typing `/head` though as I feel more confident doing so (sometimes I type the wrong number of `#`).

I also think the `/head` popup could suggest different levels of heading based on the current document structure. That is, if the last heading is level 2, we can show heading level 1, 2 and 3. I think we can safely omit 4, 5 and 6 in this case.
